docs/apache-airflow/public-airflow-interface.rst

```diff
@@ -0,0 +1,110 @@
+  a DAG, such as their dependencies and execution status.
+* `Operator <concepts/operators>`_ APIs, which allows the developers to write their custom Operators.
+* `Decorators <howto/create-custom-decorator>`_ APIs, which allows the developers to write their
+  custom decorators to make it easier to write `TaskFlow <tutorial/taskflow>`_ DAGs.
+* `Secret Managers <security/secrets>`_ APIs, which allows the developers to write their custom
+  Secret Managers to safely access credentials and other secret configuration of their workflows.
+* `Connection management <concepts/connections>`_ APIs, which allow developers to manage
+  connections to external systems
+* `XCom <concepts/xcoms>`_, which allow developers to manage cross-task communication within Airflow.
+* `Variables <concepts/variables>`_, which allow developers to manage variables within Airflow.
+* `Executors <executor/index>`_, which allow developers to manage the execution of tasks within Airflow.
+* `Listeners <listeners>`_ - all of which are considered Public Interfaces.
+* `Plugins <plugins>`_, which allow developers to extend internal Airflow capabilities - add new UI
+  pages, custom `TimeTables <concepts/timetable>`_, `Extra Links <howto/define_extra_link>`_,
+
+What is not part of the Public Interface of Apache Airflow?
+===========================================================
+
+Everything not mentioned in this document should be considered as non-Public Interface.
+
+The users however, might have some assumptions about different parts of Airflow, thinking that
+they can rely on them, so here we try to clarify and want to be explicit that they are not part of the
+Public Interface:
+
+* `Database structure <database-erd-ref>`_ is considered to be an internal implementation
+  detail and you should not assume the structure is going to be maintained in
+  backwards-compatible way.
+
+* `Web UI <ui>`_ is considered to be continuously adapting and evolving for Apache Airflow and
+  you should not assume that the UI will remain as it is in backwards-compatible way. Views and screens
+  might be updated and removed in major releases without impacting backwards-compatibility.
+
+* `Python API <python-api-ref>`_ (except the explicitly mentioned classes below), are considered an
+  internal implementation details and you should not assume they will be maintained
+  in backwards-compatible way.
+
+Which classes and packages are part of the Public Interface of Apache Airflow?
+==============================================================================
+
+The Public Interface of Airflow consists of a number of different classes and packages that provide access
+to the core features and functionality of the system.
+
+The classes and packages that may be considered as the Public Interface include:
+
+* The :class:`~airflow.models.dag.DAG`, which provides a way to define and manage DAGs in Airflow.
```

Comment:
For DAG yes. For others we would have to move it. I think - if you think it makes sense, you can add them to "airflow"  - no problem for me since we have the STAICA_HACK to avoid circular references

For now I just want to document what we have rather than  change anything.